### PR TITLE
fix(SkyApm.Diagnostics.SqlClient): added support for Microsoft.Data.SqlClient in .NET Core

### PR DESCRIPTION
**Pull request by cnlangzi**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
 #378 
___
### Bug fix
- The diagnostic name of SqlClient has been changed in Microsoft.Data.SqlClient for .NET Core. SqlClientDiagnosticProcessor does not work for them.

- Added new diagnostic listener for Microsoft.Data.SqlClient

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

File: src/SkyApm.Diagnostics.SqlClient/SqlClientDiagnosticProcessor.cs [hunk 1]
```diff
@@ -75,5 +77,43 @@ public void ErrorExecuteCommand([Property(Name = "Exception")] Exception ex)
                 _tracingContext.Release(context);
             }   
         }
+        #endregion
+
+        #region Microsoft.Data.SqlClient
+        //https://docs.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient?view=sqlclient-dotnet-core-2.1
+
+        [DiagnosticName(SqlClientDiagnosticStrings.SqlBeforeExecuteCommand)]
```
Comment (snakorse) [inline src/SkyApm.Diagnostics.SqlClient/SqlClientDiagnosticProcessor.cs]: Here should be the `DotNetCoreSqlBeforeExecuteCommand` I think?

Comment (cnlangzi) [reply src/SkyApm.Diagnostics.SqlClient/SqlClientDiagnosticProcessor.cs]: yes. it has been updated in 2nd commit.  

https://github.com/SkyAPM/SkyAPM-dotnet/pull/387/commits/ef5048b9cb8073dbebefe3ed01d4dd750133f048

The file you checked is marked as Outdated.

**Comment by snakorse** [comment]
Why there is no reference to the `Microsoft.Data.SqlClient` package? Please do some tests and make sure it realy works.

**Comment by snakorse** [comment]
I found there is a possibility that we could remove both the dependency on `Microsoft.Data.SqlClient` and `System.Data.SqlClient` packages, if we change the `SqlCommand` to `IDbCommand`, thus we will also solve problems like #304. A little problem came up with that is, we should retrieve the DataSource property on our self.

**Comment by cnlangzi** [comment]
> I found there is a possibility that we could remove both the dependency on `Microsoft.Data.SqlClient` and `System.Data.SqlClient` packages, if we change the `SqlCommand` to `IDbCommand` which is defined in netstandard2.0, thus we will also solve problems like #304. A little problem came up with that is, we should retrieve the DataSource property on our self.

May it is mistake? It works based on DiagnosticSource. `Microsoft.Data.SqlClient` and `System.Data.SqlClient` packages should be not required to import here. For #304, it may be fixed by removing `System.Data.SqlClient` reference from SkyAPM-dotnet. Let us give it test.

**Comment by snakorse** [comment]
I have test on it, `Microsoft.Data.SqlClien`t and `System.Data.SqlClient` should be referenced explictly is because currently we are 
explictly using the `SqlCommand` class which is defined in these two packages. And I just find that we can just simply replace the `SqlCommand` to `DbCommand` then the explictly dependency can be moved, the `DbCommand` is also a type defined in netstandard similar to `IDbCommand`.

**Comment by cnlangzi** [comment]
@snakorse  let me make some test. Because it works in my project without `Microsoft.Data.SqlClient` reference. Its project solution is :
- MyProject: It has `Microsoft.Data.SqlClient` reference , because our business need it to access MSSQL serer
- MyDiagnostics: A new SqlClientTracingDiagnosticProcessor is implemented for `Microsoft.Data.SqlClient`. But it has not any reference to `Microsoft.Data.SqlClient` assembly.  Of course, this project is imported by MyProject.
 It works properly in skywalking dashboard. All command is traced properly.

I will double check it, and try to make more test. I will update you if I get more results.

**Comment by cnlangzi** [comment]
But your idea is better to replace SqlCommand with IDbCommand. #304 can be fixed.

**Comment by snakorse** [comment]
I believe in your own `SqlClientTracingDiagnosticProcessor` class, you are using the `Microsoft.Data.SqlClient.SqlCommand` type for the `BeforeDotnetCoreExecuteCommand` method, but in the pr, you are actually using `System.Data.SqlClient.SqlCommand` for the method, thus this diagnostic method will never be actually called because type unmatch.

I now recommand replace SqlCommand to DbCommand instead of IDbCommand, DbCommand need less changes.

**Comment by cnlangzi** [comment]
@snakorse  Yes, you are right. Something is missed when the changes are moved from my project to fork. I have got it updated and tested pass. please check it again. thanks.

- fixed #304 : update SqlCommand with DbCommand. And "System.Data.SqlClient" has been removed from project.
- fixed #378 :  refactor code for better reading.

**Comment by snakorse** [comment]
LGTM, thanks for your work.

**Review by snakorse — APPROVED** [review]

**Pull request merged**